### ISSUE TITLE
[clang-tidy] False Positive in misc-unused-using-decls with user-declared literals

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#53444]
clang-tidy triggers a false positive here:

```cpp
#include <chrono>

using std::literals::chrono_literals::operator""s;

int main()
{
    auto x = 123s;
}
```
```
source>:3:39: warning: using decl 'operator""s' is unused [misc-unused-using-decls]
using std::literals::chrono_literals::operator""s;
                                      ^
<source>:3:39: note: remove the using
using std::literals::chrono_literals::operator""s;
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
```
https://godbolt.org/z/WaToc4q5j
